src/containerapp/azext_containerapp/_cloud_build_utils.py

```diff
@@ -0,0 +1,221 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from threading import Thread
+import os
+import re
+import requests
+import tempfile
+import time
+
+from ._archive_utils import archive_source_code
+
+from azure.cli.core.azclierror import (
+    ValidationError
+)
+
+from ._clients import BuilderClient, BuildClient
+
+from ._utils import (
+    log_in_file,
+    remove_ansi_characters
+)
+
+class CloudBuildError(Exception):
+    pass
+
+def run_cloud_build(cmd, image_name, source, location, subscription_id, resource_group_name, environment_name, run_full_id, logs_file, logs_file_path):
```

Comment:
`image_name` and `subscription_id` not been used. Should we remove them?

Reply:
This came from a previous version of the code. Thanks for catching!